File: shserc_service/src/main/java/com/dianda/shserc/entity/Resource.java
package com.dianda.shserc.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.dianda.common.entity.BaseEntity;
import com.dianda.common.util.cache.dictionary.DictionaryCache;
import com.dianda.shserc.common.Constant;
import lombok.Data;
import lombok.EqualsAndHashCode;


import java.time.LocalDateTime;

@Data
@EqualsAndHashCode(callSuper = false)
public class Resource extends BaseEntity {
	
	private String title;

	private String author;

	private String keywords;

	private String description;

	private String coverPath;

	private String resourcePath;

	private String fileExtName;

	private String fileName;

	private long fileSize;

	private boolean isLink;

	private String authorCompany;

	private String authorAddress;

	private String authorEmail;

	private String authorPhone;

	private long deformityId;

	private long resourceTypeId;

	private long mediumTypeId;

	private long stateId;

	private int viewCount;

	private long auditor;

	private String auditorOpinion;

	private LocalDateTime auditorTime;

	@TableField(exist = false)
	private String deformity;

	@TableField(exist = false)
	private String resourceType;

	@TableField(exist = false)
	private String mediumType;

	@TableField(exist = false)
	private String state;

	//20210226
	private int downloadCount;

	private int favoriteCount;

	private int commentCount;

	private int starCount;

	private int starTotal;

	private long subjectId;

	private long gradeId;

	private long stageId;

	private long importSourceId;

	private long schoolTypeId;

	private long domainId;
	
	private long learnFromId;

	@TableField(exist = false)
	private String subject;

	@TableField(exist = false)
	private String grade;

	@TableField(exist = false)
	private String stage;

	@TableField(exist = false)
	private String importSource;

	@TableField(exist = false)
	private String domain;

	@TableField(exist = false)
	private String schoolType;
	
	@TableField(exist = false)
	private  String learnFrom;

	public void addViewCount() {
		this.setViewCount(this.getViewCount() + 1);
	}

	public void addDownloadCount() {
		this.setDownloadCount(this.getDownloadCount() + 1);
	}

	public void addFavoriteCount() {
		this.setFavoriteCount(this.getFavoriteCount() + 1);
	}

	public void deleteFavoriteCount() {
		this.setFavoriteCount(this.getFavoriteCount() - 1);
	}

	public void addCommentCount() {
		this.setCommentCount(this.getCommentCount() + 1);
	}

	public void addStarCount(int score) {
		this.setStarCount(this.getStarCount() + 1);
		this.setStarTotal(this.getStarTotal() + score);
	}

	public static void dictTranslate(Resource resource, DictionaryCache cache) {
		resource.deformity = cache.translate("deformity", String.valueOf(resource.getDeformityId()), Constant.Dictonary.All );
		resource.state = cache.translate("state", String.valueOf(resource.getStateId()), Constant.Dictonary.All );
		resource.domain = cache.translate("domain", String.valueOf(resource.getDomainId()), Constant.Dictonary.All );
		resource.importSource = cache.translate("importSource", String.valueOf(resource.getImportSourceId()), Constant.Dictonary.All );
		resource.schoolType = cache.translate("schoolType", String.valueOf(resource.getSchoolTypeId()), Constant.Dictonary.All );
		resource.stage = cache.translate("stage", String.valueOf(resource.getStageId()), Constant.Dictonary.All );
		resource.grade = cache.translate("grade", String.valueOf(resource.getGradeId()), Constant.Dictonary.All );
		resource.learnFrom=cache.translate("learnFrom", String.valueOf(resource.getLearnFromId ()), Constant.Dictonary.All );
		resource.resourceType=cache.translate("resourceType", String.valueOf(resource.getResourceTypeId ()), Constant.Dictonary.All );
		resource.subject=cache.translate("subject", String.valueOf(resource.getSubjectId ()), Constant.Dictonary.All );
	}
}

File: shserc_site/src/app/models/resource.js
import { CDN } from "@/static/CDN"

export const resourceModel = {
    id: 0,
    title: "资源名称",
    state: "",
    stateId: 10002,
    deformityId: "",
    deformity: "",
    author: "",
    authorEmail: "",
    authorCompany: "",
    authorPhone: "",
    keywords: "",
    description: "",
    resourcePath: "",
    fileName: "",
    fileExtName: "",
    fileSize: 0,
    domain: "",
    domainId: 0,
    mediumType: "",
    mediumTypeId: 0,
    resourceType: "",
    resourceTypeId: 0,
    domainId: 0,
    learnFrom: "",
    learnFromId: 0,
    schoolType: "",
    schoolTypeId: 0,
    subject: "",
    subjectId: 0,
    stage: "",
    stageId: 0,
    grade: "",
    gradeId: 0,
    coverPath: CDN.DEFAULT_COVER,
    viewCount: 0,
    downloadCount: 0,
    favoriteCount: 0,
    commentCount: 0,
    isVedio: false,
    isFile: false

};

export const auditModel = {
    resourceId: 0,
    stateId: 0,
    auditorId: 0,
    auditOpinion: ""
}

export const selectParam = {

    current: 1,
    total: 0,
    size: 10,
    searchPhrase: "",

    userId: 0,
    resourceId: 0,
    deformityId: 0,
    stateId: 0,
    subjectId: 0,
    domainId: 0,
    importSourceId: 0,
    gradeId: 0,
    stageId: 0,
    resourceTypeId: 0,
    orderPhrases: {}
}

export const orderPhrasesModel = {
    favoriteCount: "favorite",
    viewCount: "view",
    downloadCount: "download"
}


//  video optons only for resouce 
export function videoOptions(videoPath, videoCover) {
    return {
        playbackRates: [0.5, 1.0, 1.5, 2.0], //可选择的播放速度
        autoplay: false, //如果true,浏览器准备好时开始回放。
        muted: false, // 默认情况下将会消除任何音频。
        loop: false, // 视频一结束就重新开始。
        preload: "auto", // 建议浏览器在<video>加载元素后是否应该开始下载视频数据。auto浏览器选择最佳行为,立即开始加载视频（如果浏览器支持）
        language: "zh-CN",
        aspectRatio: "16:9", // 将播放器置于流畅模式，并在计算播放器的动态大小时使用该值。值应该代表一个比例 - 用冒号分隔的两个数字（例如"16:9"或"4:3"）
        fluid: true, // 当true时，Video.js player将拥有流体大小。换句话说，它将按比例缩放以适应其容器。
        sources: [{
            type: "",
            src: videoPath
                //"http://shlll.s.wcsapi.biz.matocloud.com/upload201909261929354303.mp4", //url地址
        }, ],
        poster: videoCover, //你的封面地址
        // width: document.documentElement.clientWidth,
        notSupportedMessage: "此视频暂无法播放，请稍后再试", //允许覆盖Video.js无法播放媒体源时显示的默认信息。
        controlBar: {
            timeDivider: true, //当前时间和持续时间的分隔符
            durationDisplay: true, //显示持续时间
            remainingTimeDisplay: false, //是否显示剩余时间功能
            fullscreenToggle: true, //全屏按钮
        }
    }
}